torch/special/__init__.py

```diff
@@ -159,6 +159,37 @@
     tensor([-0.9466,  2.6352,  0.6131, -1.7169,  0.6261])
 """.format(**common_args))
 
+logsumexp = _add_docstr(_special.special_logsumexp,
+                        r"""
+logsumexp(input, dim, keepdim=False, *, out=None)
+
+Returns the log of summed exponentials of each row of the :attr:`input`
+tensor in the given dimension :attr:`dim`. The computation is numerically
+stabilized.
+
+For summation index :math:`j` given by `dim` and other indices :math:`i`, the result is
+
+    .. math::
+        \text{{logsumexp}}(x)_{{i}} = \log \sum_j \exp(x_{{ij}})
+
+{keepdim_details}
+
+Args:
+    {input}
+    {dim}
+    {keepdim}
+
+Keyword args:
+    {out}
+
+
+Example::
+
+    >>> a = torch.randn(3, 3)
+    >>> torch.logsumexp(a, 1)
```

Comment:
This should be `torch.special.logsumexp`. 

It would also be more meaningful to write here:
> \>\>\> torch.dist(torch.special.logsumexp(a, 1), a.exp().sum(1).log())
> tensor(1.1921e-07)

Reply:
I'm unfortunately on a different opinion with this change, I'm thinking that we should be consistent with examples on other functions and I feel having `>>> torch.special.logsumexp(a, 1)` is OK here.

Okay, I think your suggestion also makes sense, so to satisfy both simplicity as well as give a better meaning to what the function does. How about this instead?

```python
>>> torch.special.logsumexp(a, 1)
#output
>>> torch.dist(torch.special.logsumexp(a, 1), torch.log(torch.sum(torch.exp(a), dim)))
#output
```

Reply:
I think having both would be a good adition. Note that the formula I wrote had an errata, and  `dim` should be `1`.